wolfcrypt/benchmark/benchmark.c

```diff
@@ -6404,6 +6422,97 @@ void bench_sakke(void)
 #endif /* WOLFCRYPT_SAKKE_CLIENT */
 #endif /* WOLFCRYPT_HAVE_SAKKE */
 
+#ifdef HAVE_LIBOQS
+void bench_falconKeySign(byte level)
+{
+    int    ret = 0;
+    falcon_key key;
+    double start;
+    int    i, count;
+    byte   sig1[FALCON_LEVEL1_SIG_SIZE];
+    byte   sig5[FALCON_LEVEL5_SIG_SIZE];
+    byte   msg[512];
+    word32 x = 0;
+    const char**desc = bench_desc_words[lng_index];
+
+    ret = wc_falcon_init(&key);
+    if (ret != 0) {
+        printf("wc_falcon_init failed\n");
+        return;
+    }
+
+    ret = wc_falcon_set_level(&key, level);
+    if (ret != 0) {
```

Comment:
The code needs refactored to make sure `wc_falcon_free(&key);` will always be called after `wc_falcon_init` succeeds.

Reply:
Memory management bites me again! Will do it.

Reply:
oh right...I remember now...I got careless because I thought this is just benchmarking code.